app/lib/system/deploy.rb

```diff
@@ -4,28 +4,44 @@ module System
   module Deploy
     mattr_accessor :info
 
+    DEFAULT_DEPLOY_INFO_PATH = '.deploy_info'
+    private_constant :DEFAULT_DEPLOY_INFO_PATH
+
     # Provides information about the version of the code:
     #   - revision: low-level version, that is used by BugSnag as app version
     #   - release: customer-facing version, that is visible in the admin portal footer
     #   - deployed_at: timestamp of the deployment
-    # The data is taken from the `.deploy_info` file in the root directory, but for SaaS the release
-    # is overridden with VERSION.
+    #   - docs_version: product name and the version that are used as prefix for product documentation path in the
+    #                   Red Hat Customer Portal
+    # The data is taken from the `.deploy_info2` file in the root directory, but for SaaS the release
+    # is overridden with DEFAULT_VERSION.
+    # `.deploy_info2` file is injected to the container during container build process in CPaaS (the content is set in Dockerfile)
     # The information is exposed via {MASTER_PORTAL}/deploy endpoint for logged-in users.
     class Info
-      VERSION = '2.x'
-      private_constant :VERSION
+      DEFAULT_VERSION = '2.x'
+
+      private_constant :DEFAULT_VERSION
 
       attr_reader :revision, :deployed_at, :release
 
       delegate :minor_version, :major_version, to: :version
 
       def initialize(info)
         @revision = info.fetch('revision') { `git rev-parse HEAD 2> /dev/null`.strip }
-        @release = ThreeScale.config.onpremises ? info.fetch('release', VERSION) : VERSION
+        @release = ThreeScale.saas? ? DEFAULT_VERSION : info.fetch('release', DEFAULT_VERSION)
         @deployed_at = info.fetch('deployed_at') { Time.now }
         @error = info.fetch(:error) if info.key?(:error)
       end
 
+      def docs_version
+        @docs_version ||= ThreeScale.saas? || rhoam? ? 'red_hat_3scale/2-saas' : "red_hat_3scale_api_management/#{major_version}.#{minor_version}"
+      end
+
+      # RHOAM version has only one segment (release = 'RHOAM')
+      def rhoam?
```

Comment:
Yes, I agree it's confusing, but here is some reasoning:

1. I first added `saas?` in this module as well, but then CodeClimate complained (something like "method doesn't depend on this instance state, consider moving to another class"), and I thought the suggestion made sense and moved it to `three_scale.rb` https://github.com/3scale/porta/pull/3540/commits/2022bf800c9a69920a5c05121d887b8a2af7da5b. This also made it possible to reuse this method in other places, e.g. remove the same method from `application_helper.rb`.

Now there are also a bunch of places where we can replace `ThreeScale.config.onpremises` with `!saas?`, but I decided to keep it for another PR (as it might be error-prone, and it's pure refactoring, not adding anything).

2. While `saas?` is (or potentially will be) used in many places throughout the code, this `rhoam?` thing so far is needed just here, in order to provider the docs base URL. It also depends on what is parsed from the file in this Deploy module, so it's the only place where it makes sense.